extensions/ql-vscode/src/data-extensions-editor/model-details/model-details-panel.ts

```diff
@@ -17,7 +18,10 @@ export class ModelDetailsPanel extends DisposableObject {
     this.push(treeView);
   }
 
-  public setExternalApiUsages(externalApiUsages: ExternalApiUsage[]): void {
-    this.dataProvider.setExternalApiUsages(externalApiUsages);
+  public setState(
+    externalApiUsages: ExternalApiUsage[],
+    databaseItem: DatabaseItem,
+  ): void {
+    this.dataProvider.setState(externalApiUsages, databaseItem);
```

Comment:
This surprised me a little bit. I've generally thought of the "DataProvider" classes as the ones managing state and expected the data extensions editor view to use the provider rather than the panel. Having said that, I can't think of a problem with what you've done here 🤔

Reply:
Cool! Lets go with that and see if we can identify any pros/cons about either approach. In general it feels like it would be good to standardise a bit to make the codebase a bit easier.